src/ls/Unit_LS.ml

```diff
@@ -0,0 +1,250 @@
+        {
+          str = "";
+          pos =
+            {
+              charpos = String.length "print(\"hello world\")";
+              line = 1;
+              column = 0;
+              file;
+            };
+        } )
+    in
+    {
+      Pattern_match.rule_id;
+      file;
+      range_loc;
+      tokens = Lazy.from_val [];
+      env = [];
+      taint_trace = None;
+      engine_kind = Pattern_match.OSS;
+    }
+  in
+  let matches = Common.map match_of_file files in
+  (matches, [ rule ])
+
+let mock_workspace ?(git = false) () =
+  let rand_dir () =
+    let rand_int_string = Random.int 100000 |> string_of_int in
+    let dir_name = "test_workspace_" ^ rand_int_string in
+    let dir = Filename.concat (Filename.get_temp_dir_name ()) dir_name in
+    Unix.mkdir dir 0o777;
+    dir
+  in
+  let workspace =
+    try rand_dir () with
+    | _ -> rand_dir ()
+  in
+  Unix.chdir workspace;
+  if git then Sys.command "git init" |> ignore;
+  workspace
+
+let add_file ?(git = false) ?(dirty = false)
+    ?(content = "print(\"hello world\")\n") workspace () =
+  let file = Filename.temp_file ~temp_dir:workspace "test" ".py" in
+  let oc = open_out_bin file in
+  output_string oc content;
+  close_out oc;
+  let add_file file = Sys.command ("git add " ^ file) |> ignore in
+  let commmit () = Sys.command "git commit -m 'test'" |> ignore in
+  if git then add_file file;
+  if (not dirty) && git then commmit ();
+  file
+
+let session_targets () =
+  let test_session files expected root only_git_dirty =
+    let session = mock_session () in
+    let session = { session with only_git_dirty; root } in
+    let session = set_session_targets session files in
+    let%lwt { target_mappings; _ } = Session.targets session in
+    let targets =
+      Common.map (fun target -> target.Input_to_core_t.path) target_mappings
+    in
+    let targets = Common.sort targets in
+    let expected = Common.sort expected in
+    Alcotest.(check (list string)) "targets" expected targets;
+    Lwt.return_unit
+  in
+  let test_no_git () =
+    let workspace = mock_workspace () in
+    let file1 = add_file workspace () in
+    let file2 = add_file workspace () in
+    let files = [ file1; file2 ] in
+    let expected = files in
+    test_session files expected workspace false
+  in
+  let test_git_dirty_no_git () =
+    let workspace = mock_workspace () in
+    let file1 = add_file workspace () in
+    let file2 = add_file workspace () in
+    let files = [ file1; file2 ] in
+    let expected = files in
+    test_session files expected workspace true
+  in
+  let test_git () =
+    let workspace = mock_workspace ~git:true () in
+    let file1 = add_file ~git:true workspace () in
+    let file2 = add_file ~git:false workspace () in
+    let files = [ file1; file2 ] in
+    let expected = files in
+    test_session files expected workspace false
+  in
+  let test_git_dirty () =
+    let workspace = mock_workspace ~git:true () in
```

Comment:
@mjambon started to write some test utilities in help creating test files and dirs.
See libs/Testutil_files.mli

Reply:
Maybe we need to extend it to support those ~git:xxx flags.